components/relevancy/src/populate_interests.rs

```diff
@@ -161,4 +220,76 @@ mod test {
         })
         .unwrap();
     }
+
+    #[test]
+
+    fn test_parse_records_with_bad_domain_strings() {
+        viaduct_reqwest::use_reqwest_backend();
+        let records_string = r#"
+        {
+            "id": "animals-0001",
+            "type": "category_to_domains",
+            "record_custom_details": {
+              "category_to_domains": {
+                "category": "animals",
+                "category_code": 1,
+                "version": 1
+              }
+            }
+        }"#;
+        let attachment_string = r#"
+        [
+            {"domain": "badString"},
+            {"domain": "notBase64"}
+        ]"#;
+
+        let record_fields: RelevancyRecord = serde_json::from_str(records_string).unwrap();
+        let parsed_attachment_data: Vec<RelevancyAttachmentData> =
+            serde_json::from_str(attachment_string).unwrap();
+        let mut data: Vec<(Interest, UrlHash)> = vec![];
+        parse_record(record_fields, parsed_attachment_data, &mut data)
+            .expect_err("Invalid base64 error");
+    }
+
+    #[test]
+
+    fn test_parse_records() {
+        viaduct_reqwest::use_reqwest_backend();
+        let records_string = r#"
+        {
+            "id": "animals-0001",
+            "type": "category_to_domains",
+            "record_custom_details": {
+              "category_to_domains": {
+                "category": "animals",
+                "category_code": 1,
+                "version": 1
+              }
+            }
+        }"#;
+        let attachment_string = r#"
+        [
+            {"domain": "uKCZ+w36B1JgqmAfbi7n+g=="},
+            {"domain": "BFHucYqTr/Df7BfJ72gUjg=="}
+        ]"#;
+
+        let record_fields: RelevancyRecord = serde_json::from_str(records_string).unwrap();
+        let parsed_attachment_data: Vec<RelevancyAttachmentData> =
+            serde_json::from_str(attachment_string).unwrap();
+        let mut data: Vec<(Interest, UrlHash)> = vec![];
+        parse_record(record_fields, parsed_attachment_data, &mut data).unwrap();
+        assert_eq!(
+            data,
+            vec![
+                (
+                    Interest::Animals,
+                    [184, 160, 153, 251, 13, 250, 7, 82, 96, 170, 96, 31, 110, 46, 231, 250]
```

Comment:
Nit: maybe replace this with `hash_url("the-url").unwrap()`.  I think that will be easier to understand later on.